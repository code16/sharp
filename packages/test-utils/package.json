{
    "name": "@sharp/test-utils",
    "private": true,
    "main": "src/index.js",
    "version": "1.0.0"
}