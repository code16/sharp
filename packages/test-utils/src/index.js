export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC, unmockSFC } from './mockSFC';
export { mockProperty, unmockProperty, setter } from './mock-utils';
export { nextRequestFulfilled } from './moxios-utils';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}