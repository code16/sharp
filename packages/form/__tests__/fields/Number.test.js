import Vue from 'vue';
import NumberInput from '../../src/components/fields/Number.vue';

import { MockInjections } from '@sharp/test-utils';


describe('number-field',()=>{
    Vue.component('sharp-number', NumberInput);

    beforeEach(()=>{
        document.body.innerHTML = `
            <div id="app">
                <sharp-number placeholder="Entrez un nombre" value="1" :show-controls="false" step="1" min="0" max="10"></sharp-number>
            </div>
        `
    });

    test('can mount Number field', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});


async function createVm(customOptions={}) {
    const vm = new Vue({
        el: '#app',
        mixins: [MockInjections, customOptions],
    });

    await Vue.nextTick();

    return vm.$children[0];
}