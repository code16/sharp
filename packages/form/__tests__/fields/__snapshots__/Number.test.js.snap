// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`number-field can mount Number field 1`] = `
<div id="app">
	<input type="number" placeholder="Entrez un nombre" class="SharpText hide-controls" step="1" min="0" max="10">
</div>
`;
